### PR TITLE
[RFR] Api public and Api functional tests

### PR DESCRIPTION
**Pull request by alexisjanvier**

### CODE REVIEW

File: e2e/api/public/api.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/* eslint func-names:0 no-undef:0*/
```
Comment (jpetitcolas) [inline e2e/api/public/api.js]: Can we instead add `describe` and `it` to the global available variables in `.eslintrc`?

File: e2e/api/public/apiAuthenticate.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
+/* eslint func-names:0 no-undef:0*/
+
+describe.only('/api/authenticate', () => {
```
Comment (jpetitcolas) [inline e2e/api/public/apiAuthenticate.js]: Only spotted

File: src/api/lib/db/queries/index.js [hunk 3]
```diff
@@ -47,7 +47,7 @@ const base = (client, tableName, exposedFields, searchableFieldsList, sortableFi
 
     return {
         selectAll: select.selectAll,
-        selectOne: select.selectOneById,
+        selectOne: select.selectOneById, // Not sure that's a great id !!!
```
Comment (Kmaschta) [inline src/api/lib/db/queries/index.js]: :+1:

File: src/api/lib/middlewares/pgCrud.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+import koaRoute from 'koa-route';
+
+export default (queriesFactory, availableMethods = ['GET', 'POST', 'PUT', 'DELETE']) => {
+    const app = koa();
+    let queries;
+
+    app.use(function* (next) {
+        const clean = method => method.toLowerCase().trim();
+        const method = clean(this.method);
+        this.isAllowed = availableMethods.map(clean).indexOf(method) !== -1;
```
Comment (Kmaschta) [inline src/api/lib/middlewares/pgCrud.js]: I suggest to rename this attribute `methodIsAllowed` if it's accessible on all next middlewares.

File: e2e/api/public/apiProducts.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                url: '/api/products',
+            });
+            assert.equal(statusCode, 200);
+        });
+        it('should return a list of all products', function* () {
+            const { body } = yield request({
+                method: 'GET',
+                url: '/api/products',
+            });
+            assert.equal(body.length, 3);
```
Comment (Kmaschta) [inline e2e/api/public/apiProducts.js]: It could be nice to test the three products' integrity ?

File: src/api/lib/db/queries/index.js (hunk 3 above)
Comment (jpetitcolas) [inline src/api/lib/db/queries/index.js]: :+1: for the comment pun!

I don't think however we should create some aliases. This is a boilerplate after all. New devs should learn repositories names. :)

**Comment by Kmaschta** [comment]
Need rebase

File: e2e/api/public/api.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+/* eslint func-names:0 */
```
Comment (Kmaschta) [inline e2e/api/public/api.js]: Like globals in `.eslintrc`, I think we could put the func-names rules in it too.

Comment (alexisjanvier) [inline e2e/api/public/api.js]: Yes, I hesitated ... At the same time it's a good practice, but it's a pain in tests ...
@djhi @jpetitcolas an opinion ?

File: e2e/api/public/apiProducts.js (hunk 5 above)
Comment (alexisjanvier) [inline e2e/api/public/apiProducts.js]: done

File: e2e/api/public/api.js (hunk 6 above)
Comment (djhi) [inline e2e/api/public/api.js]: It's a pain for generators too but I guess this should still be enabled. We might however add an `.eslintrc` inside the `e2e` folder which extends the root one

Comment (alexisjanvier) [inline e2e/api/public/api.js]: :+1:  i'll try that

**Pull request merged**